### PR TITLE
AN-24 Adding support for gallery captions

### LINKED ISSUE #97: Caption support for galleries
Do galleries using [gallery] shortcode get exported to Apple News with captions supported?

### PR DESCRIPTION
**Pull request by kevinfodness**
* Adds support for captions on images in galleries.
* Adds support for alt text on images in galleries.
* Adds a unit test for complex galleries including both captions and alt text.
* Refactors the `Gallery` and `Gallery_Test` classes for adherence to WordPress standards and PHP best practices.

Fixes #97.

### CODE REVIEW

File: includes/apple-exporter/components/class-gallery.php [hunk 1]
```diff
@@ -13,56 +28,100 @@ class Gallery extends Component {
...
+
+			// Add the compiled slide content to the list of items.
+			$items[] = $content;
 		}
 
+		// Build the JSON.
 		$this->json = array(
-			'role'   => $this->get_setting( 'gallery_type' ),
-			'items'  => $items,
+			'role' => $this->get_setting( 'gallery_type' ),
```
Comment (tomharrigan) [inline includes/apple-exporter/components/class-gallery.php]: yes. I dislike tab spacing alignment. 1 space is the way to go :)

File: includes/apple-exporter/components/class-gallery.php [hunk 2]
```diff
@@ -13,56 +28,100 @@ class Gallery extends Component {
...
-		$this->register_full_width_layout( 'gallery-layout', array(
-			'margin' => array( 'top' => 25, 'bottom' => 25 )
-		) );
+		$this->register_full_width_layout(
+			'gallery-layout',
+			array(
+				'margin' => array(
+					'bottom' => 25,
+					'top' => 25,
+				)
```
Comment (tomharrigan) [inline includes/apple-exporter/components/class-gallery.php]: trailing comma

File: tests/apple-exporter/components/test-class-gallery.php [hunk 3]
```diff
@@ -1,67 +1,144 @@
...
 
+	/**
+	 * Test the apple_news_gallery_json filter.
+	 *
+	 * @access public
+	 */
+	public function testFilter() {
+
+		// Setup.
+		$html = <<<HTML
```
Comment (tomharrigan) [inline tests/apple-exporter/components/test-class-gallery.php]: this block exists in 3 places, worth abstracting?

Comment (kevinfodness) [inline tests/apple-exporter/components/test-class-gallery.php]: Makes sense to me. I'll do that.

**Pull request merged**